src/test/java/io/confluent/idesidecar/restapi/kafkarest/RecordsV3Suite.java

```diff
@@ -199,17 +199,21 @@ static ArgumentSets validKeysAndValues() {
   @CartesianTest
   @CartesianTest.MethodFactory("validKeysAndValues")
   default void testProduceAndConsumeData(RecordData key, RecordData value) {
-    var topicName = randomTopicName();
+    produceAndConsume(this, key, value);
+  }
+
+  static <T extends ITSuite> void produceAndConsume(T test, RecordData key, RecordData value) {
```

Comment:
This is used in the `RecordsV3WithoutSRSuite`.

Reply:
This is probably okay, but don't you think we probably want a common interface that defines the common (non-static) methods that could be used in `RecordsV3Suite` and `RecordsV3WithoutSRSuite`?